driver-core/src/test/functional/com/mongodb/internal/operation/AsyncCommandBatchCursorFunctionalTest.java

```diff
@@ -103,6 +106,95 @@ void cleanup() {
         });
     }
 
+    @Test
+    @DisplayName("should exhaust cursor with multiple batches")
+    void shouldExhaustCursorAsyncWithMultipleBatches() {
+        // given
+        BsonDocument commandResult = executeFindCommand(0, 3); // Fetch in batches of size 3
+        cursor = new AsyncCommandBatchCursor<>(TimeoutMode.CURSOR_LIFETIME, commandResult, 3, 0, DOCUMENT_DECODER,
+                null, connectionSource, connection);
+
+        CompletableFuture<List<List<Document>>> futureResult = new CompletableFuture<>();
+
+       // when
+        cursor.exhaustCursor((result, throwable) -> {
+            if (throwable != null) {
+                futureResult.completeExceptionally(throwable);
+            } else {
+                futureResult.complete(result);
+            }
+        });
```

Comment:
Let's use `FutureResultCallback` here and in `shouldExhaustCursorAsyncWithClosedCursor`, `shouldExhaustCursorAsyncWithEmptyCursor`.